packages/sdk/src/core/classes/contract-wrapper.ts

```diff
@@ -450,6 +450,11 @@ export class ContractWrapper<
       gas = BigNumber.from(500000);
     }
 
+    // check for gas override in callOverrides
+    if (BigNumber.from(callOverrides?.gasLimit).gt(gas)) {
```

Comment:
can calloverrides be undefined? I think it should always be there. 

`gasLimit` can be undefined though, and im not sure what happens if you call `BigNumber.from(undefined)` - might want to add a `if (callOverrdes.gasLimit && BigNumber.from(callOverrides?.gasLimit).gt(gas)` to be safe

Reply:
makes sense, updating